diff --git a/src/cmake/Config.cmake.in b/src/cmake/Config.cmake.in
index beac9bc..2b5638a 100644
--- a/src/cmake/Config.cmake.in
+++ b/src/cmake/Config.cmake.in
@@ -36,6 +36,8 @@ endif()
 if (NOT @OPENIMAGEIO_CONFIG_DO_NOT_FIND_IMATH@ AND NOT OPENIMAGEIO_CONFIG_DO_NOT_FIND_IMATH)
     find_dependency(Imath @Imath_VERSION@
                     HINTS @Imath_DIR@)
+    find_dependency(OpenEXR @OpenEXR_VERSION@
+                        HINTS @OpenEXR_DIR@)
 endif ()
 
 if (NOT @fmt_LOCAL_BUILD@ AND (NOT @OIIO_INTERNALIZE_FMT@ OR @OIIO_USE_COMPILED_FMT@))
@@ -127,7 +129,7 @@ endif ()
 
 # Compute the installation prefix relative to this file. Note that cmake files are installed
 # to ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME} (see OIIO_CONFIG_INSTALL_DIR)
-get_filename_component(_CURR_INSTALL_LIBDIR "${CMAKE_CURRENT_LIST_DIR}/../" ABSOLUTE)
+get_filename_component(_CURR_INSTALL_LIBDIR "${CMAKE_CURRENT_LIST_DIR}/../" ABSOLUTE)
 get_filename_component(_ABS_CMAKE_INSTALL_LIBDIR "@CMAKE_INSTALL_FULL_LIBDIR@" ABSOLUTE)
 get_filename_component(_ABS_CMAKE_INSTALL_INCLUDEDIR "@CMAKE_INSTALL_FULL_INCLUDEDIR@" ABSOLUTE)
 file(RELATIVE_PATH _INCLUDEDIR_RELATIVE_TO_LIBDIR
